crates/rolldown_common/src/chunk/mod.rs

```diff
@@ -219,9 +210,16 @@ impl Chunk {
       }
     });
 
-    let filename = filename_template.render(Some(&name), None, hash_replacer);
+    let filename = filename_template.render(Some(chunk_name), None, hash_replacer).into();
+
+    let name = if has_hash_pattern {
+      make_unique_name(&filename, used_name_counts);
```

Comment:
You could extract it before the ifStmt, since it will exec anyway

Reply:
Yeah. Here could be improved, here not change it because i want to make pr minimal.

Reply:
Let's review it at https://github.com/rolldown/rolldown/pull/4195.